src/cfml/system/endpoints/Maven.cfc

```diff
@@ -0,0 +1,701 @@
+			var thisArtifactPath = artifactService.getArtifactPath( artifactParts.artifactId, artifactParts.version );
+
+			// Return the path to the artifact
+			return fileEndpoint.resolvePackage(
+				thisArtifactPath,
+				currentWorkingDirectory,
+				arguments.verbose
+			);
+		}
+
+		// Check only the explicitly defined repo, if any
+		if ( len( artifactParts.repo ) ) {
+			artifact = getArtifactFromRepo(
+				artifactParts.repo,
+				artifactParts.groupId,
+				artifactParts.artifactId,
+				artifactParts.version
+			);
+			jarFileURL = artifact.jarFileURL;
+		}
+		// Otherwise, check each registered repo sequentially
+		else {
+			for ( var alias in repos ) {
+				artifact = getArtifactFromRepo(
+					repos[ alias ],
+					artifactParts.groupId,
+					artifactParts.artifactId,
+					artifactParts.version
+				);
+				jarFileURL = artifact.jarFileURL;
+				// If we found the artifact, break out of the loop
+				if ( jarFileURL.len() ) {
+					break;
+				}
+			}
+		}
+
+		// Defer to jar endpoint
+		var folderName = jarEndpoint.resolvePackage(
+			artifact.jarFileURL,
+			currentWorkingDirectory,
+			arguments.verbose
+		);
+
+		if ( artifactParts.version eq "STABLE" ) {
+			artifactParts.version = getLatestVersion( artifactParts.groupId, artifactParts.artifactId );
+		}
+
+		// Update artifact version if it's a range
+		else if ( !semanticVersion.isExactVersion( artifactParts.version, true ) ) {
+			if (
+				artifact.artifactMetadata.keyExists( "versioning" ) && artifact.artifactMetadata.versioning.keyExists( "versions" ) && artifact.artifactMetadata.versioning.versions.len()
+			) {
+				var sortedVersions = artifact.artifactMetadata.versioning.versions.sort( ( a, b ) => variables.semanticVersion.compare( b, a ) );
+				// Get the latest version that matches the range
+				for ( var thisVersion in sortedVersions ) {
+					if ( semanticVersion.satisfies( thisVersion, artifactParts.version ) ) {
+						artifactParts.version = thisVersion;
+						break;
+					}
+				}
+			}
+		}
+
+		// get dependencies
+		var artifactDependencies = getArtifactAndDependencyJarURLs(
+			artifactParts.groupId,
+			artifactParts.artifactId,
+			artifactParts.version
+		);
+
+		var installPaths = {};
+		var dependencies = {};
+
+		for ( var dependency in artifactDependencies ) {
+			if ( dependency.artifactId == artifactParts.artifactId ) {
+				continue;
+			}
+			dependencies[ dependency.artifactId ] = getNamePrefixes() & (
+				artifactParts.repo.len() ? artifactParts.repo & "|" : ""
+			) & dependency.groupId & ":" & dependency.artifactId & ":" & dependency.version;
+			installPaths[ dependency.artifactId ] = "lib/" & dependency.artifactId;
+		}
+
+		// override the box.json with the actual version and dependencies
+		var boxJSON = {
+			"name"         : "#artifactParts.groupId & "-" & artifactParts.artifactId#.jar",
+			"slug"         : artifactParts.artifactId,
+			"version"      : artifactParts.version,
```

Comment:
This should be the actual version that was installed, not the incoming semver range

Reply:
that is actually the installed version. My logic updates the artifacts version above when resolving the range and I use it here.